AntaresSimulatorTeam/AntaREST
# feat: add unit tests and improve source code documentation

## Description

This pull request introduces several enhancements to the codebase related to exporting study data. The changes primarily focus on adding comprehensive unit tests for various methods and improving documentation to make the code more maintainable and robust.

## Changes Made

1. Added unit tests for the `generate_task` method in `VariantStudyService`.
2. Implemented unit tests for the `export_study_flat` method in `RawStudyService`.
3. Introduced unit tests for the `export_study` method in `AbstractStorageService`.
4. Resolved an issue related to study export handling scenarios with mixed folders and ZIP files in the output directory.
5. Created fixtures in `conftest.py` to set up services required for testing.
6. Refactored unit tests by separating helper functions from fixtures in `conftest.py`.
7. Added missing docstring to the `storage_service` module for better code documentation.
8. Corrected a spelling mistake in the `variant_study_service` parameter of the `StudyStorageService`.
9. Removed unused fixtures (database) from the export tests.
10. Improved the documentation for the `AbstractStorageService.export_study` method.

## How to Test

1. Clone this branch to your local environment.
2. Run the unit tests using the testing framework of your choice (e.g., pytest).
3. Verify that all tests pass successfully and cover the relevant scenarios.
4. Manually test the study export functionality with a mix of folders and ZIP files.
